homeassistant/components/media_player/denon.py

```diff
@@ -116,32 +116,57 @@ def telnet_request(cls, telnet, command, all_lines=False):
 
     def telnet_command(self, command):
         """Establish a telnet connection and sends `command`."""
-        telnet = telnetlib.Telnet(self._host)
-        _LOGGER.debug("Sending: %s", command)
+        try:
+            _LOGGER.debug("%s: Opening telnet for sending command",
+                          self.entity_id)
+            telnet = telnetlib.Telnet(self._host, 23, DEFAULT_TIMEOUT)
```

Comment:
This should be moved to a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#6-communication-with-devices-services

Reply:
@MartinHjelmare ,

I was just doing this to resolve an issue with an existing component. I understand that overall guidance is to have devices in separate libraries but there should be exceptions to this when the PR is for an already existing component with putting in a small fix.
Doing what you ask would mean rewrite of this component to fix an issue.

Reply:
The issue is with the interface code. So it's a good time to move this to a library and fix it there instead. We shouldn't keep fixing interface code that shouldn't be in home assistant in the first place.